### PR TITLE
Add v5 guide for PRs by Erika

### PR DESCRIPTION
**Pull request by Princesseuh**
#### Related issues & labels (optional)

- Closes #
- Suggested label:

### CODE REVIEW

**Comment by sarah11918** [comment]
Thanks @Princesseuh !

I know there will need to be changes to the Adapter API reference page too from some of your code changes, so I haven't yet done that and will leave that for you.

But, I did make some tiny changes related to `compiledContent()` and  no more Squoosh support in [this commit](https://github.com/withastro/docs/pull/9240/commits/6c6798dd11d245a4f6193ce7a392e31275febe52) that it would be great if you could just edit directly if I didn't get them right.

I think after this PR, the API adapter page, and that commit, you'll officially be off docs duty! 🫡

File: src/content/docs/en/guides/upgrade-to/v5.mdx [hunk 1]
```diff
@@ -1002,8 +1002,60 @@ If you were using the `assets` property, remove this as it is no longer availabl
 
 You may also wish to update your supported features with the new `limited` option and include a message about your adapter's support.
 
+```ts title="my-adapter.mjs" del={1-5} ins={7-11}
```
Comment (sarah11918) [inline src/content/docs/en/guides/upgrade-to/v5.mdx:1005]:
```suggestion
```ts title="my-adapter.mjs" del={2-5} ins={7-10}
```

File: src/content/docs/en/guides/upgrade-to/v5.mdx [hunk 2]
```diff
@@ -1002,8 +1002,60 @@ If you were using the `assets` property, remove this as it is no longer availabl
...
+
 <ReadMore>Read more about [specifying supported Astro features in an adapter](/en/reference/adapter-reference/#astro-features).</ReadMore>
 
+### Removed: Deprecated definition shape for dev toolbar apps (Dev Toolbar API)
+
+<SourcePR number="11987" title="Remove deprecated dev toolbar app shape"/>
+
+Previously, the `addDevToolbarApp` method inside integrations allowed defining an app by using the `addDevToolbarApp(string);` signature and then in the entrypoint of the toolbar app, the default export would have the `id`, `title`, and `icon` properties to define the app.
+
+This shape was deprecated in Astro 4.x and has now been fully removed in Astro 5.0, in favor of a new shape that's more intuitive and allows Astro to provide better errors when toolbar apps fails to load correctly.
```
Comment (sarah11918) [inline src/content/docs/en/guides/upgrade-to/v5.mdx:1027]:
```suggestion
In Astro 4.x, when building a dev toolbar app, it was still possible to use the previously deprecated `addDevToolbarApp(string);` signature. The `id`, `title`, and `icon` properties to define the app were then made available through the default export of the app's `entrypoint`.

Astro 5.0 completely removes this option entirely in favor of the current object shape when defining a dev toolbar app in an integration that's more intuitive and allows Astro to provide better errors when toolbar apps fail to load correctly.
```

**Review by sarah11918 — COMMENTED** [review]
Looks fabulous @Princesseuh ! Just updated some highlighted line numbers and did the 4.x, 5.0 thing for your new entry.

Happy for you to merge this when you're ready!

**Comment by Princesseuh** [comment]
Updated the adapter reference with the new changes. I checked the compiledContent and Squoosh changes and they looked fine to me!

**Pull request merged**